addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/PolyfillCdnScriptScanRule.java

```diff
@@ -101,12 +128,22 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 // Check the script contents, in case they are loading scripts via JS
                 for (Element sourceElement : sourceElements) {
                     String contents = sourceElement.getContent().toString();
-
-                    for (Pattern pattern : ALL_DOMAINS) {
-                        Matcher matcher = pattern.matcher(contents);
-                        if (matcher.find()) {
-                            this.createLowConfidenceAlert(null, matcher.group(0)).raise();
-                            break;
+                    String contentsLc = contents.toLowerCase();
+
+                    Matcher urlMatcher = ALL_URLS.matcher(contents);
```

Comment:
Did you try without matching first for URLs? That seems to be equally fast.

Reply:
Tried it and yes, it is as fast 😁 Reverted to not extract the URLs as it makes the code simpler.